contracts/liquidators/SolidlyLiquidator.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.0;
+
+import "openzeppelin-contracts-upgradeable/contracts/token/ERC20/IERC20Upgradeable.sol";
+import "openzeppelin-contracts-upgradeable/contracts/token/ERC20/utils/SafeERC20Upgradeable.sol";
+
+import { IRouter } from "../external/solidly/IRouter.sol";
+
+import "./IRedemptionStrategy.sol";
+
+/**
+ * @title SolidlyLiquidator
+ * @notice Exchanges seized token collateral for underlying tokens via a Solidly router for use as a step in a liquidation.
+ */
+contract SolidlyLiquidator is IRedemptionStrategy {
+  using SafeERC20Upgradeable for IERC20Upgradeable;
+
+  /**
+   * @notice Redeems custom collateral `token` for an underlying token.
+   * @param inputToken The input wrapped token to be redeemed for an underlying token.
+   * @param inputAmount The amount of the input wrapped token to be redeemed for an underlying token.
+   * @param strategyData The ABI-encoded data to be used in the redemption strategy logic.
+   * @return outputToken The underlying ERC20 token outputted.
+   * @return outputAmount The quantity of underlying tokens outputted.
+   */
+  function redeem(
+    IERC20Upgradeable inputToken,
+    uint256 inputAmount,
+    bytes memory strategyData
+  ) external override returns (IERC20Upgradeable outputToken, uint256 outputAmount) {
+    return _convert(inputToken, inputAmount, strategyData);
+  }
+
+  function _convert(
+    IERC20Upgradeable inputToken,
+    uint256 inputAmount,
+    bytes memory strategyData
+  ) internal returns (IERC20Upgradeable outputToken, uint256 outputAmount) {
+    // Get Solidly router and path
+    (IRouter solidlyRouter, address tokenTo) = abi.decode(strategyData, (IRouter, address));
+
+    bool stable;
+
+    // Check if stable pair exists
+    address volatilePair = solidlyRouter.pairFor(address(inputToken), tokenTo, false);
+    address stablePair = solidlyRouter.pairFor(address(inputToken), tokenTo, true);
+
+    if (!solidlyRouter.isPair(stablePair)) {
+      stable = false;
+    } else if (!solidlyRouter.isPair(volatilePair)) {
+      stable = true;
+    } else {
+      (uint256 stableR0, uint256 stableR1) = solidlyRouter.getReserves(address(inputToken), tokenTo, true);
+      (uint256 volatileR0, uint256 volatileR1) = solidlyRouter.getReserves(address(inputToken), tokenTo, false);
```

Comment:
it doesn't look like this works for non-pairs 
![image](https://user-images.githubusercontent.com/2662912/235203510-e6f72da4-ec11-4d3a-8c07-7e8bb57187b7.png)